pkg/gui/controllers/helpers/app_status_helper.go

```diff
@@ -66,3 +78,21 @@ func (self *AppStatusHelper) renderAppStatus() {
 		}
 	})
 }
+
+func (self *AppStatusHelper) renderAppStatusSync(stop chan struct{}) {
+	ticker := time.NewTicker(time.Millisecond * 50)
+	go func() {
+	outer:
+		for {
+			select {
+			case <-ticker.C:
+				appStatus := self.statusMgr().GetStatusString()
+				self.c.SetViewContent(self.c.Views().AppStatus, appStatus)
+				_ = self.c.GocuiGui().ForceRedrawView(self.c.Views().AppStatus)
```

Comment:
Redrawing just the AppStatus view only works when it doesn't get shorter during the operation. Which is the case for our use of WithWaitingStatusSync, but maybe it's not general enough?

Reply:
Redrawing just the AppStatus is not enough: it only overwrites the options view instead of pushing it to the right.

This one is a bit tricky to solve; I added a fixup that redraws both the AppStatus view and the Options view, but it also has to freeze the text for the Information view to prevent it from changing to "rebasing" during the operation. It's a little bit ugly, but does the job.

Reply:
In the new version we redraw all views of the bottom line, this works well. Also, we tell the ModeHelper to disregard the rebasing mode while we're doing the rebase, so that the information view doesn't flicker. This is better than the previous way of freezing the information string.